pkgs/by-name/sl/slint-viewer/package.nix

```diff
@@ -0,0 +1,45 @@
+{
+  lib,
+  rustPlatform,
+  fetchCrate,
+  qt6,
+  libGL,
+  nix-update-script,
+  versionCheckHook,
+}:
+rustPlatform.buildRustPackage (finalAttrs: {
+  pname = "slint-viewer";
+  version = "1.11.0";
+
+  src = fetchCrate {
+    inherit (finalAttrs) pname version;
+    hash = "sha256-Yez8GbER6ylkozQP5oQ0m0u+x/T5qQVPRt0S/NRFT60=";
+  };
+
+  useFetchCargoVendor = true;
+  cargoHash = "sha256-vWTj6cJgvg10NaLw9WfHXmiG8hg7mUIH/Gj3JVvWCuA=";
+
+  buildInputs = [
+    qt6.qtbase
+    qt6.qtsvg
+    libGL
+  ];
+
+  nativeBuildInputs = [ qt6.wrapQtAppsHook ];
+
+  # There are no tests
+  doCheck = false;
+  doInstallCheck = true;
+  nativeInstallCheckInputs = [ versionCheckHook ];
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    description = "Viewer for .slint files from the Slint Project";
+    mainProgram = "slint-viewer";
+    homepage = "https://crates.io/crates/slint-viewer";
+    license = lib.licenses.gpl3Only;
+    platforms = lib.platforms.linux ++ lib.platforms.darwin;
```

Comment:
The github page also mentions that Windows is supported. Current it doesn't build in nixpkgs due to Python 3.12 being broken for cross though, but that's not slint's fault.
https://github.com/NixOS/nixpkgs/blob/1e0230733b61c776d840f0fdcae0ca0b703c75f6/pkgs/development/interpreters/python/cpython/default.nix#L827-L830

Reply:
Oh, I didn't know nix would package anything at all for windows. Thanks for the suggestion.